packages/shared/routes/onboarding/views/stronghold-setup/views/SetupStrongholdPasswordView.svelte

```diff
@@ -38,24 +36,39 @@
                 errKey = `error.password.${passwordInfo[passwordStrength?.feedback.warning]}`
             }
             error = localize(errKey)
-        } else if (password !== confirmedPassword) {
+        } else if (strongholdPassword !== confirmedStrongholdPassword) {
             errorConfirm = localize('error.password.doNotMatch')
         } else {
             try {
                 busy = true
-                if (existingPassword) {
-                    // TODO: also pass in currentPassword
-                    await changeStrongholdPassword(password)
+
+                const mustChangePassword =
+                    $onboardingProfile?.strongholdPassword &&
+                    $onboardingProfile?.strongholdPassword !== strongholdPassword
+                if (mustChangePassword) {
+                    await changeStrongholdPassword($onboardingProfile?.strongholdPassword, strongholdPassword)
+
+                    if ($iotaProfileManager) {
+                        await $iotaProfileManager.changeStrongholdPassword(
+                            $onboardingProfile?.strongholdPassword,
+                            strongholdPassword
+                        )
+                    }
```

Comment:
Why do we create the iota manager so soon, If its only used in claiming? Would mean we don't need to add all of this logic in previous flows